### PR TITLE
Prune action nodes that were already computed and add demo

### PR DESCRIPTION
**Pull request by shravan97**
In this PR : 
* Prune action nodes that were already computed
* Add unit tests for the pruning
* Add a demo notebook

### CODE REVIEW

File: PyRDF/Node.py [hunk 1]
```diff
@@ -194,4 +194,13 @@ def graph_prune(self):
             
             return False
 
+        elif not self.children and self.value:
```
Comment (etejedor) [inline PyRDF/Node.py]: `not self.children` is checked twice in this if-elif, can you reorganize the conditions to check self.children only once at the beginning?

File: PyRDF/Node.py [hunk 2]
```diff
@@ -194,4 +194,13 @@ def graph_prune(self):
...
+
+            ### If the current node's value was already
+            ### computed, it should get pruned only if it's
+            ### an action node.
+
+            if self.operation and self.operation.is_action():
```
Comment (etejedor) [inline PyRDF/Node.py]: Isn't this condition doing the contrary than what one would expect? Here you are **not** pruning if it is an action that already has a value, and that is precisely what you **do** want to prune, right?

Comment (shravan97) [reply PyRDF/Node.py]: @etejedor , there's actually some confusion here. It looks like I'm pruning the nodes only when `graph_prune` returns `false`. But I've documented it the opposite way. I'm fixing this right away 👍

**Comment by shravan97** [comment]
@etejedor , changes made and rebased too 👍

**Pull request merged**